### PR TITLE
[Fix] Refine focal loss

### PR DESCRIPTION
**Pull request by juncaipeng**
image

### CODE REVIEW

File: paddleseg/models/losses/focal_loss.py [hunk 1]
```diff
@@ -23,38 +23,61 @@
...
+            logit (Tensor): Logit tensor, the data type is float32, float64. Shape is
+                (N, C, H, W), where C is number of classes.
+            label (Tensor): Label tensor, the data type is int64. Shape is (N, W, W),
+                where each value is 0 <= label[i] <= C-1.
+        Returns:
+            (Tensor): The average loss.
+        """
+        assert logit.ndim == 4, "The ndim of logit should be 4."
+        assert logit.shape[1] == 2, "The channel of logit should be 2."
+        assert label.ndim == 3, "The ndim of loglabelit should be 3."
```
Comment (wuyefeilin) [inline paddleseg/models/losses/focal_loss.py]: loglabelit  拼写错误

File: paddleseg/models/losses/focal_loss.py [hunk 2]
```diff
@@ -23,38 +23,61 @@
...
+
+        loss = F.sigmoid_focal_loss(
+            logit=logit,
+            label=label,
+            alpha=self.alpha,
+            gamma=self.gamma,
+            reduction='none')
+        loss = loss * mask
+        avg_loss = paddle.sum(loss) / paddle.sum(
+            paddle.cast(loss != 0., 'int32'))
```
Comment (wuyefeilin) [inline paddleseg/models/losses/focal_loss.py]: 这边是否可以直接用mask作为分母呢，另外需要防止分母为0的情况

File: paddleseg/core/train.py [hunk 3]
```diff
@@ -42,11 +42,7 @@ def loss_computation(logits_list, labels, losses, edges=None):
         loss_i = losses['types'][i]
         coef_i = losses['coef'][i]
 
-        if loss_i.__class__.__name__ in ('BCELoss', 'FocalLoss'
-                                         ) and loss_i.edge_label:
-            # If use edges as labels According to loss type.
-            loss_list.append(coef_i * loss_i(logits, edges))
-        elif loss_i.__class__.__name__ == 'MixedLoss':
```
Comment (wuyefeilin) [inline paddleseg/core/train.py:49]: 这个删掉不会有影像吗

File: paddleseg/models/losses/focal_loss.py (hunk 1 above)
Comment (juncaipeng) [reply paddleseg/models/losses/focal_loss.py]: OK

File: paddleseg/models/losses/focal_loss.py (hunk 2 above)
Comment (juncaipeng) [reply paddleseg/models/losses/focal_loss.py]: 主要是直观一点，我加上 EPS 避免error。

File: paddleseg/core/train.py (hunk 3 above)
Comment (juncaipeng) [reply paddleseg/core/train.py:49]: 嗯，在这两个loss实现中都没有使用edge_label这个参数，是无用的参数。

**Review by wuyefeilin — APPROVED** [review]
LGTM

**Review by michaelowenliu — APPROVED** [review]
